webapp/src/components/assetEditor/store/assetEditorReducer.ts

```diff
@@ -76,20 +76,35 @@ function getUserAssets() {
 
     const animationToGalleryItem = (asset: pxt.Animation) => {
         if (asset.frames?.length <= 0) return null;
-        let bitmap = pxt.sprite.Bitmap.fromData(asset.frames[0]);
-        asset.previewURI = imgConv.convert("data:image/x-mkcd-f," + pxt.sprite.base64EncodeBitmap(bitmap.data()));
+        asset.framePreviewURIs = asset.frames.map(bitmap => imgConv.convert("data:image/x-mkcd-f," + pxt.sprite.base64EncodeBitmap(bitmap)));
+        asset.previewURI = asset.framePreviewURIs[0];
         return asset;
     };
 
-    const images = project.getAssets(pxt.AssetType.Image).map(imageToGalleryItem).sort(compareInternalId);
-    const tiles = project.getAssets(pxt.AssetType.Tile).map(imageToGalleryItem)
-        .filter(t => !t.id.match(/^myTiles.transparency(8|16|32)$/gi)).sort(compareInternalId);
-    const tilemaps = project.getAssets(pxt.AssetType.Tilemap).map(tilemapToGalleryItem).sort(compareInternalId);
-    const animations = project.getAssets(pxt.AssetType.Animation).map(animationToGalleryItem);
-
-    const assets = images.concat(tiles).concat(tilemaps).concat(animations);
+    const getAssetType = gallery ? project.getGalleryAssets.bind(project) : project.getAssets.bind(project);
+
+    const images = getAssetType(pxt.AssetType.Image).map(imageToGalleryItem).sort(compareInternalId);
+    const tiles = getAssetType(pxt.AssetType.Tile).map(imageToGalleryItem)
+        .filter((t: pxt.Tile) => !t.id.match(/^myTiles.transparency(8|16|32)$/gi)).sort(compareInternalId);
+    const tilemaps = getAssetType(pxt.AssetType.Tilemap).map(tilemapToGalleryItem).sort(compareInternalId);
+    const animations = getAssetType(pxt.AssetType.Animation).map(animationToGalleryItem);
```

Comment:
I see there wasn't one before, but should there be a `.sort(compareInternalId)` here too?